packages/devtools_app/lib/src/screens/memory/panes/allocation_tracing/allocation_profile_class_table.dart

```diff
@@ -0,0 +1,138 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import '../../../../primitives/utils.dart';
+import '../../../../shared/common_widgets.dart';
+import '../../../../shared/table.dart';
+import '../../../../shared/table_data.dart';
+import '../../../../shared/utils.dart';
+import '../../memory_controller.dart';
+import 'allocation_profile_tracing_view_controller.dart';
+
+/// The default width for columns containing *mostly* numeric data (e.g.,
+/// instances, memory).
+const _defaultNumberFieldWidth = 80.0;
+
+class _TraceCheckBoxColumn extends ColumnData<TracedClass>
+    implements ColumnRenderer<TracedClass> {
+  _TraceCheckBoxColumn()
+      : super(
+          'Trace',
+          titleTooltip:
+              'Enable or disable allocation tracing for a specific type',
+          fixedWidthPx: scaleByFontFactor(55.0),
+          alignment: ColumnAlignment.left,
+        );
+
+  @override
+  bool get supportsSorting => false;
+
+  @override
+  Widget build(
+    BuildContext context,
+    TracedClass item, {
+    bool isRowSelected = false,
+    VoidCallback? onPressed,
+  }) {
+    final controller =
+        Provider.of<MemoryController>(context).allocationTracingController;
+    return Checkbox(
+      value: controller.isAllocationTracingEnabledForClass(item.cls),
+      onChanged: (value) async {
+        await controller.setAllocationTracingForClass(item.cls, value!);
```

Comment:
After checking 'Text', and then finding it on top and selecting the row (for hello world flutter application on mac), got red screen with error: 

```
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following _CastError was thrown building ValueListenableBuilder<TracedClass?>(dirty, state:
_ValueListenableBuilderState<TracedClass?>#07e51):
Null check operator used on a null value

The relevant error-causing widget was:
  ValueListenableBuilder<TracedClass?>
  ValueListenableBuilder:file:///Users/polinach/_/devtools/packages/devtools_app/lib/src/screens/memory/panes/allocation_tracing/allocat
  ion_profile_tracing_tree.dart:29:12

When the exception was thrown, this was the stack:
#0      AllocationTracingTree.build.<anonymous closure>
(package:devtools_app/src/screens/memory/panes/allocation_tracing/allocation_profile_tracing_tree.dart:46:57)
#1      _ValueListenableBuilderState.build (package:flutter/src/widgets/value_listenable_builder.dart:186:26)
#2      StatefulElement.build (package:flutter/src/widgets/framework.dart:4981:27)
#3      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4868:15)
#4      StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5034:11)
#5      Element.rebuild (package:flutter/src/widgets/framework.dart:4595:5)
#6      BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2672:19)
#7      WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:882:21)
#8      RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:374:5)
#9      SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1175:15)
#10     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1104:9)
#11     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1015:5)
#15     _invoke (dart:ui/hooks.dart:148:10)
#16     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:316:5)
#17     _drawFrame (dart:ui/hooks.dart:113:31)
(elided 3 frames from dart:async)
```